tests/Functional/RsyncFileCopierTest.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace PhpTuf\ComposerStager\Tests\Functional;
+
+use PhpTuf\ComposerStager\Exception\DirectoryNotFoundException;
+use PhpTuf\ComposerStager\Infrastructure\Process\FileCopier\RsyncFileCopier;
+use Symfony\Component\Process\ExecutableFinder;
+
+/**
+ * @coversDefaultClass \PhpTuf\ComposerStager\Infrastructure\Process\FileCopier\RsyncFileCopier
+ * @covers ::__construct
+ * @covers ::copy
+ * @uses \PhpTuf\ComposerStager\Infrastructure\Filesystem\Filesystem
+ * @uses \PhpTuf\ComposerStager\Infrastructure\Process\ExecutableFinder
+ * @uses \PhpTuf\ComposerStager\Infrastructure\Process\ProcessFactory
+ * @uses \PhpTuf\ComposerStager\Infrastructure\Process\Runner\AbstractRunner
+ */
+class RsyncFileCopierTest extends TestCase
+{
+    public static function setUpBeforeClass(): void
+    {
+        if (!self::isRsyncAvailable()) {
+            return;
+        }
+        self::createTestEnvironment();
+    }
+
+    public static function tearDownAfterClass(): void
+    {
+        if (!self::isRsyncAvailable()) {
+            return;
+        }
+        self::removeTestEnvironment();
+    }
+
+    protected function setUp(): void
+    {
+        if (!self::isRsyncAvailable()) {
+            $this->markTestSkipped('Rsync is not available for testing.');
+        }
+    }
+
+    protected static function isRsyncAvailable(): bool
+    {
+        $finder = new ExecutableFinder();
+        return !($finder->find('rsync') === null);
```

Comment:
Wouldn't something like this be a little clearer?
```suggestion
        return $finder->find('rsync') !== null;
```

Reply:
Yes, I think so.